src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -753,12 +753,17 @@ void ServiceBasedNodeInfoAccessor::HandleNotification(const GcsNodeInfo &node_in
     // was alive and is now dead or resources have been updated.
     bool was_alive = (entry->second.state() == GcsNodeInfo::ALIVE);
     is_notif_new = was_alive && !is_alive;
-    // Once a node with a given ID has been removed, it should never be added
-    // again. If the entry was in the cache and the node was deleted, check
-    // that this new notification is not an insertion.
-    if (!was_alive) {
-      RAY_CHECK(!is_alive)
-          << "Notification for addition of a node that was already removed:" << node_id;
+    
+    // When a new node(node-B) registers with GCS, it subscribes to all node information.
+    // It will subscribe to redis and then get all node information from GCS through RPC.
+    // If node-A fails after GCS replies to node-B, GCS will send another message(node-A
+    // is dead) to node-B through redis publish. Because RPC and redis subscribe are two
+    // different sessions, node-B may process node-A dead message first and then node-A
+    // alive message.
+    if (!was_alive && is_alive) {
+      RAY_LOG(INFO) << "Notification for addition of a node that was already removed:"
```

Comment:
Better to be a warning.

Reply:
https://docs.google.com/document/d/1i5HYUnmsBjd35U8LprD0PyUDoPjVJUP4ucA4assxSHU/edit#heading=h.dowskc9eehq9 According to the ray logging policy, INFO is better.

Reply:
I see.

Reply:
Yeah. We are trying to have a better logging policy on OSS ray. 

WARNING or ERROR should better have "action items" unless it scares users.
As a new policy, I am thinking to add INFO shouldn't be used to explain implementation details. This is because people who will look at INFO logs will not be familiar with ray codebase.